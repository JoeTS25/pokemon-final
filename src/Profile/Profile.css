@font-face {
    font-family: pokemon-solid;
    src: url('../Pokemon\ Solid.ttf');
}

@font-face {
  font-family: pokemon;
  src: url('../Pokemon\ Classic.ttf');
}

.head2 {
  font-family: pokemon-solid;
  color: rgb(225, 44, 44);
  padding-top: 50px;
}

.container {
    padding-left: 150px;
}

.battle {
  background-color: rgb(213, 222, 209);
  height: 250px;
  width: 400px;
  padding-top: 20px;
  padding-left: 20px;
}

.p1, .p2 {
  font-family: pokemon;
}

.p1 {
  margin-bottom: 10px;
}

.btn {
  width: 100px;
  height: 25px;
}

.move-link {
  font-family: pokemon-solid;
  color: white;
}

.link-div {
  height: 30px;
  padding-top: 20px;
}