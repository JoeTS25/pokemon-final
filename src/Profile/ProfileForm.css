@font-face {
    font-family: pokemon-solid;
    src: url('../Pokemon\ Solid.ttf');
}

.head-3 {
    font-family: pokemon-solid;
    padding-left: 25px;
    color: rgb(241, 219, 76);
    margin-top: 10px;
}

.user-label {
    font-family: pokemon-solid;
    padding-left: 30px;
    margin-left: 6px;
    background-color: rgba(219, 36, 48, 0.935);
}

.email-label {
    margin-right: 10px;
    margin-left: 30px;
    font-family: pokemon-solid;
    background-color: rgba(219, 36, 48, 0.935);
}

.password-label {
    margin-left: 30px;
    margin-right: 10px;
    font-family: pokemon-solid;
    background-color: rgba(239, 12, 28, 0.935);
}

.btn {
    margin-left: 30px;
    font-family: pokemon-solid;
    background-color: rgba(219, 36, 48, 0.935);
}