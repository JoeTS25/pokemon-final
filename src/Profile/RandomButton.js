import React from "react";

function RandomButton({ handleClick }) {
   return (
    <div className="rand-button">
        <button className="btn" onClick={handleClick}><b>Battle</b></button>
    </div>
   )
}

export default RandomButton;