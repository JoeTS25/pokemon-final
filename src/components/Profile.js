import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";

function Profile() {
    const { currentUser } = useContext(UserContext);
    console.debug("Profile", "currentUser=", currentUser);

    return (
        <div className="Profile">
            <h1>{currentUser.username}'s Team</h1>
            {currentUser
                ? <h2>Current Team</h2>
                : (
                    <p>
                        <Link to="/">Go back home</Link>
                    </p>
                )}

        </div>
    )
}

export default Profile;