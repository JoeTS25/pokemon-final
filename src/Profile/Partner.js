import React, { useEffect, useState } from "react";
import axios from "axios";
import { Link } from "react-router-dom";
import RandomButton from "./RandomButton";
import CapName from "../Hooks/Capitalize";

function Partner({ currentUser }) {
   const randAtt = Math.floor(Math.random() * 11);
   const [randomId, setRandomId] = useState([randAtt]);
   const pokeId = `https://pokeapi.co/api/v2/pokemon/${randomId}`;
   const [currentPoke, setCurrentPoke] = useState([]);
   const [nextPoke, setNextPoke] = useState();
   const [nextAtt, setNextAtt] = useState();
   const [newDivOne, setnewDivOne] = useState();
   const [newDivTwo, setNewDivTwo] = useState();
   const [newPic, setNewPic] = useState();
   const [pokeLink, setPokeLink] = useState();
   const [moveLink, setMoveLink] = useState();
   const [isLoading, setIsLoading] = useState(true);
   
   async function GetPoke() {
    setIsLoading(true);
    const res = await axios.get(pokeId);
    setCurrentPoke(poke => {
            poke = [...poke, res.data]
            return poke
        });
    setNextPoke(Math.floor(Math.random() * 1026));
    setNextAtt(Math.floor(Math.random() * 11));
    setIsLoading(false);
   };
   

   useEffect(() => {
    GetPoke();
   }, [pokeId]);

   const pokeData = currentPoke[0];

   function handleClick(evt) {
    evt.preventDefault();
    setCurrentPoke([]);
    setRandomId(nextPoke);
    setNextAtt(Math.floor(Math.random() * 21));
    
    setnewDivOne(
        `${currentUser.username} sent out a`
    )
    
    setPokeLink(
        CapName(pokeData.name)
    )
    
    setNewDivTwo(
        `${CapName(pokeData.name)} used ${CapName(pokeData.moves[nextAtt].move.name)}`
    )
    
    setNewPic(pokeData.sprites.front_default)

    setMoveLink(`/move/${pokeData.moves[nextAtt].move.name}`)

  }

  



   if (isLoading) return ("Loading...");


   return(
    <div className="poke-p">
        <div className="battle">
            <div className="p1-div">
                <p className="p1">{newDivOne} {pokeLink}</p>
            </div>
            <div className="img-div"><img src={newPic} alt="" className="img-1"/></div>
            <div className="p2-div">
                <p className="p2">{newDivTwo}</p>
            </div>
        </div>
        <div className="link-div">
        <Link to={moveLink} className="move-link">Attack Info</Link>
        </div>
        <RandomButton handleClick={handleClick}/>
        
    </div>
   )

}

export default Partner;