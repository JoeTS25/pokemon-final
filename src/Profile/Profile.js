import React, { useContext } from "react";
import UserContext from "../auth/UserContext";
import "./Profile.css";
import Partner from "./Partner";

/* User profile page. Welcomes user if logged in
*  Shows login/register if not 
*/

function Profile() {
    const { currentUser } = useContext(UserContext);
    console.debug("Profile", "currentUser=", currentUser);

    return (
        <div className="Profile">
            <div className="container text-center">
                {currentUser
                    ? <div>
                        <h2 className="head2">Welcome Back {currentUser.username}!</h2>
                    <Partner currentUser={currentUser}/>
                    </div>
                    : (
                      <h1>Must be logged in</h1>
                    )}
            </div>
        </div>
    )
}

export default Profile;