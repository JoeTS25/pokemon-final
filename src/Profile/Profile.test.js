import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Profile from "./Profile";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Profile />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider currentUser={null}>
            <Profile />
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });