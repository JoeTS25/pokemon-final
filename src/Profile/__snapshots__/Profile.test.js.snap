// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Profile"
  >
    <div
      class="container text-center"
    >
      <div>
        <h2
          class="head2"
        >
          Welcome Back testuser!
        </h2>
        Loading...
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Profile"
  >
    <div
      class="container text-center"
    >
      <h1>
        Must be logged in
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
